### PR TITLE
Remove generate.go and put the rest of it in better places.

### PR DESCRIPTION
**Pull request by pphaneuf**
### Checklist

- [ ] I have updated the [CHANGELOG](CHANGELOG.md).
- [ ] I have updated [documentation](docs/) accordingly (including the [feature implementation matrix](docs/Feature_Implementation_Matrix.md)).

### CODE REVIEW

File: crypto/keys/pem/pem_test.go [hunk 1]
```diff
@@ -21,6 +21,8 @@ import (
 	. "github.com/google/trillian/crypto/keys/pem"
 	ktestonly "github.com/google/trillian/crypto/keys/testonly"
 	"github.com/google/trillian/testonly"
+
+	_ "github.com/golang/glog"
```
Comment (NatalieDoduc) [inline crypto/keys/pem/pem_test.go:25]: why this change?

**Review by NatalieDoduc — COMMENTED** [review]
Could you please amend the commit description to say "keys/generate.go"?

I forget what it is at the moment (proto generation maybe?), but i think we also have more generic "generate.go" files and at first I was thinking you were removing that.

File: crypto/keys/pem/pem_test.go (hunk 1 above)
Comment (pphaneuf) [reply crypto/keys/pem/pem_test.go:25]: Some of those tests seem to be run by hand by the GCB scripts (in addition to being run via `go test`), and get passed some glog flags, which are not recognised without this.

This was previously pulled indirectly, somehow (not obvious to me in this PR, but there's been much code deleted), it was found via GCB failures and going through the error logs.

**Review by NatalieDoduc — APPROVED** [review]
thanks for the clarifications!
